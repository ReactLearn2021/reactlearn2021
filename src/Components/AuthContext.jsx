import React, { useState } from "react";

export const AuthContext = React.createContext();
export const AuthProvider = ({ children }) => {
    const [loggedIn, setLoggedIn] = useState(false),
          logIn = (email, password) => {
                if (email !== "actual@mail.ru" || password !== "жальКонечнЧтоТайпскриптаНет((") {
                    return;
                }

                setLoggedIn(true);
          },
          logOut = () => {
                setLoggedIn(false);
          };

    return(
        <AuthContext.Provider value = {{ logIn, logOut, loggedIn }}>
            { children }
        </AuthContext.Provider>
    )
};

export const withAuth = (Wrapped) => {
    return class extends React.Component {
        render() {
            return(
                <AuthContext.Consumer>
                    {
                        (value) => <Wrapped {...value} {...this.props} />
                    }
                </AuthContext.Consumer>
            )
        }
    }
}