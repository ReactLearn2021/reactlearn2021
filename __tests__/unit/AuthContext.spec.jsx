/**
 * @jest-environment jsdom
 */

import React from "react";
import { AuthProvider, AuthContext } from "../../src/Components/AuthContext";
import { render } from "@testing-library/react";
import { act } from "react-dom/test-utils";
import renderer from "react-test-renderer";

describe("AuthContext", () => {
    test("it sets 'loggedIn' to false value", () => {
        let loggedIn,
            logIn;

        render(
            <AuthProvider>
                <AuthContext.Consumer>
                    { (value) => {
                        loggedIn = value.loggedIn;
                        logIn = value.logIn;
                        return null;
                    } }
                </AuthContext.Consumer>
            </AuthProvider>
        );

        expect(loggedIn).toEqual(false);
        act(() => {
            logIn("actual@mail.ru", "жальКонечнЧтоТайпскриптаНет((");
        });
        expect(loggedIn).toBeTruthy();
    });

    test("it matches snapshot", () => {
        const tree = renderer.create(<AuthProvider />).toJSON();
        expect(tree).toMatchSnapshot(); // отрендерился как null
    });
});