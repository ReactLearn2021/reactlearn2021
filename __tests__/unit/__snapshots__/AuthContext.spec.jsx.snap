// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthContext it matches snapshot 1`] = `null`;
